source/architecture.rst

```diff
@@ -1,230 +1,245 @@
+MQTT for example, and send then to the context broker for internal
+distribution, reaching, for instance, the history service so it can persist the
+data on a database and the CEP for processing it based on rules. If certain
+conditions are matched when rules are being processed, a new event is generated
+and sent to the broker service to be redistributed to the interested services.
+
+More detailed and updated information can be found on the `dojot Github
+repository <https://github.com/dojot>`_.
+
+Each one of the components that are part of the architecture are briefly
+described on the sub-sections below.
+
+Kafka + data-broker + NGSI
 ***********************************
 
-Apache Kafka is a distributed messaging platform that can be used by applications which need to stream data or
-consume/produce data pipelines. In contrast to what Orion is, Kafka seems to be more appropriate to fulfil
-*dojot*'s architectural requirements (responsibility isolation, simplicity, and so on). And using it with a
-subscription manager and a NGSI interface translator, we can compose an entity which is very close to
-the features offered by Orion, in addition to improved speed and easier scalability.
+Apache Kafka is a distributed messaging platform that can be used by
+applications which need to stream data or consume/produce data pipelines. In
+comparison with other open-source messaging solutions, Kafka seems to be more
+appropriate to fulfil *dojot*'s architectural requirements (responsibility
+isolation, simplicity, and so on).
 
-In Kafka, a specialized topics structure is used to insure isolation between different users and applications data,
-enabling a multi-tenant infrastructure.
+In Kafka, a specialized topics structure is used to insure isolation between
+different users and applications data, enabling a multi-tenant infrastructure.
 
-The subscription manager service makes use of an in-memory database for efficiency. It adds context to Apache Kafka,
-making it possible that internal or even external services are able to subscribe or query data based on context. The
-subscription manager is also a distributed service to avoid it being a single point of failure or even a bottleneck for
-the architecture.
+The flow-broker service makes use of an in-memory database for efficiency. It
+adds context to Apache Kafka, making it possible that internal or even external
+services are able to subscribe or query data based on context. Flow-broker is
+also a distributed service to avoid it being a single point of failure or even
+a bottleneck for the architecture.
 
-To keep a certain level of compatibility with Fiware services, meaning, for using Fiware services and components in the
-*dojot* platform with the minimal amount of adaptations, we added a NGSI interface translation service.
+To keep a certain level of compatibility with NGSI-compatible components, we
+added a NGSI interface translation service.
```

Comment:
Who is this NGSI translation service?

Reply:
This NGSI translation thing was supposed to be a library or something that would give a NGSI interface to current components. As there isn't any component that has (or needs) such interface, I'll rewrite this paragraph - it might be interesting for a reader to know that this thing is possible.